experimental/OrthogonalDiscriminants/src/direct.jl

```diff
@@ -0,0 +1,221 @@
+  if p == 0
+    id = identifier(tbl)
+  else
+    id = identifier(ordinary_table(tbl))
+  end
+
+  # Collect candidates that may afford `chi`.
+  # Prefer an entry that knows to belong to `chi` or a member from its
+  # orbit under group automorphisms,
+  # but admit also other entries that do not know their characters,
+  # provided that characteristic and degree determine the representation
+  # up to group automorphisms.
+#TODO: Improve by ruling out candidates via their character fields.
+  d = degree(Int, chi)
+  infos1 = []
+  infos2 = []
+  orbs = orbits_group_automorphisms(tbl)
+  orb = orbs[findfirst(x -> chipos in x, orbs)]
+  unique_orbit = all(x -> chipos in x || degree(Int, tbl[x[1]]) != d, orbs)
+  for info in all_atlas_group_infos(id, characteristic => p, dim => d)
+    if haskey(info, :constituents)
+      # We trust in the precomputed data.
+      if length(info[:constituents]) == 1 && info[:constituents][1] in orb
+        push!(infos1, info)
+      end
+    elseif unique_orbit
+      # Any abs. irred. character of degree `d` will be good.
+      push!(infos2, info)
+    end
+  end
+
+  # Run over the candidates (usually just one).
+  for info in vcat(infos1, infos2)
+    G = atlas_group(info)
+    G == nothing && continue
+    F = base_ring(G)
+    K, embK = character_field(chi)
+    if degree(F) != degree(K)
+      if p == 2 || p == 0
+        # For even `p`, we need representations over the character field.
+        # Thus try to write the matrices over the character field.
+        V = free_module(F, degree(G))
+        M = gmodule(G, [hom(V, V, x) for x in [matrix(x) for x in gens(G)]])
+        Mmin = Oscar.GModuleFromGap.gmodule_minimal_field(M)
+#TODO: For p == 0, we need also the (!) embedding of `base_ring(Mmin)`
+#      into `F == base_ring(M)`, because we know the embedding of `F`
+#      into a cyclotomic field, and we will need this embedding in the end.
+        F = base_ring(Mmin)
+        (degree(F) != degree(K)) && continue
+        G = matrix_group([matrix(x) for x in Mmin.ac])
+      elseif is_even(degree(F) // degree(K))
+        # For odd p, it is sufficient that `F` is an odd degree extension
+        # of the character field.
+        V = free_module(F, degree(G))
+        M = gmodule(G, [hom(V, V, x) for x in [matrix(x) for x in gens(G)]])
+        Mmin = Oscar.GModuleFromGap.gmodule_minimal_field(M)
+        F = base_ring(Mmin)
+        is_even(degree(F) // degree(K)) && continue
+        G = matrix_group([matrix(x) for x in Mmin.ac])
+      end
+    end
+
+    if ! haskey(info, :constituents)
+      # We do not yet know that `G` has character `chi`.
+      # Check whether the representation is abs. irred.
+      if p == 0
+        # We can check abs. irred. if we can reduce the repres.
+        # modulo a prime l that does not divide the group order.
+        l = next_prime(max([x[1] for x in factor(order(tbl))]...))
+        Gbar, _ = Oscar.isomorphic_group_over_finite_field(G, min_char = Int(l))
```

Comment:
Ideally this should be 
```M = gmodule(Gbar)
is_abs_irr(M)
```
or similar. The "natural" gamodule of a matrix group is the missing bit...